src/osal/lv_freertos.c

```diff
@@ -179,8 +179,8 @@ lv_result_t lv_mutex_delete(lv_mutex_t * pxMutex)
 lv_result_t lv_thread_sync_init(lv_thread_sync_t * pxCond)
 {
 #if USE_FREERTOS_TASK_NOTIFY
-    /* Store the handle of the calling task. */
-    pxCond->xTaskToNotify = xTaskGetCurrentTaskHandle();
+    pxCond->xTaskToNotify = NULL;
```

Comment:
Please don't remove the scope of the init API.

Add a local function here like:
```
    /* If the cond is uninitialized, perform initialization. */
    prvCheckCondInit(pxCond);
```
and let the initialization happen if it get called explicitly.

Then make use of the same API `prvCheckCondInit(pxCond);` in the `lv_thread_sync_wait()`.